tests/importer/importer.cpp

```diff
@@ -1106,3 +1106,79 @@ TEST(Importer, basePathNotEndingInSlash)
     importer->resolveImports(importedUnits, resourcePath("importer"));
     EXPECT_FALSE(importedUnits->hasUnresolvedImports());
 }
+
+TEST(Importer, importInvalidXml)
+{
+    auto e = "The attempt to import the model at '" + resourcePath("importer/not_even_proper.xml") + "' failed: the file is not valid XML.";
+
+    auto importer = libcellml::Importer::create();
+    auto parser = libcellml::Parser::create();
+
+    auto model = parser->parseModel(fileContents("importer/import_invalid_xml.cellml"));
+
+    importer->resolveImports(model, resourcePath("importer"));
+    EXPECT_EQ(size_t(1), importer->errorCount());
+    EXPECT_EQ(e, importer->error(0)->description());
+}
+
+TEST(Importer, importInvalidUnitsFromCellmlModel)
+{
+    auto e = "Encountered an error when resolving units 'units1_imported' from '" + resourcePath("importer/invalid_model.cellml") + "'.";
```

Comment:
This message doesn't help me to determine what is wrong when resolving units 'units1_imported'. So, as an end-user, how would I go about fixing the problem?

Same question for tests `importInvalidComponentFromCellmlModelComponentError`, `importInvalidComponentFromCellmlModelVariableError` and `importInvalidComponentFromCellmlModelResetError`.

Reply:
A) I think it does. It tells the user that the units they are trying to import as 'units1_imported' has some problem.  The problem was noticed when trying to read in the units from the model defined by the path.
B) You would have to parse the model at the path and look at the errors given and decide if they are errors that effect your use of the model.

Reply:
Agreed with **A)**, but I don't see why I would need to do **B)**. The importer tells us that something is wrong, so I would expect to be told exactly what is wrong and not be "forced" to do **B)** to find out exactly what is wrong.

Reply:
the goal of this PR is to simply get an error when something goes wrong with a bit more information than the current `nullptr` burried in the resolved model. This seems like a good first step in that direction without needing to spend time on implementing a full propagation of error handling up model import hierarchies. This now gives the user sufficient information to track down the problem.

In future we could look to propagate all errors in way that is safe, meaningful, and robust across a hirarchy of model imports, but I don't think we need that now.